### DISCUSSION TITLE
How to make gizmos draw above everything else in 2d?

### DISCUSSION THREAD

**Discussion by brianduff** [bevyengine/bevy#9362] [Q&A] [answered]
When I draw 2d gizmos to the screen, they seem to get drawn behind my sprites, so are not visible. I think this is probably because of the z-ordering of my 2d sprites (my sprite transforms use z=1. for background tiles and z=2. for movable sprites). 

I thought that I could maybe fix this by setting the `depth_bias` of the `GizmoConfig` to `-1`, since this is documented to draw gizmos above everything else in the scene. However, this didn't seem to have any effect. I can see the gizmo being drawn if I turn off my entire scene background rendering (i.e. disable all my sprites). 

It seems like gizmos are unconditionally drawn at z-coordinate 0. in 2d irrespective of the `depth_bias`. What's the right way to make the gizmos appear above everything else in 2d?

I put together a small example demonstrating this here:
https://github.com/brianduff/gizmodepth

If you change the sprite's transform z co-ordinate in main.rs to 0. rather than 1. you can see the gizmo as expected.

**Comment by nicopap** [comment, marked as answer]
There is currently a bug that makes gizmos always draw behind everything. It will be fixed in the next hotfix release 

* See https://github.com/bevyengine/bevy/issues/9308 for a workaround and more details
